core/model/src/main/java/org/eclipse/rdf4j/model/datatypes/XMLDatatypeUtil.java

```diff
@@ -289,6 +290,78 @@ public static boolean isValidValue(String value, IRI datatype) {
+			result = isValidInteger(value);
+		} else if (datatype == CoreDatatype.XSD_NEGATIVE_INTEGER) {
+			result = isValidNegativeInteger(value);
+		} else if (datatype == CoreDatatype.XSD_NON_POSITIVE_INTEGER) {
+			result = isValidNonPositiveInteger(value);
+		} else if (datatype == CoreDatatype.XSD_NON_NEGATIVE_INTEGER) {
+			result = isValidNonNegativeInteger(value);
+		} else if (datatype == CoreDatatype.XSD_POSITIVE_INTEGER) {
+			result = isValidPositiveInteger(value);
+		} else if (datatype == CoreDatatype.XSD_LONG) {
+			result = isValidLong(value);
+		} else if (datatype == CoreDatatype.XSD_INT) {
+			result = isValidInt(value);
+		} else if (datatype == CoreDatatype.XSD_SHORT) {
+			result = isValidShort(value);
+		} else if (datatype == CoreDatatype.XSD_BYTE) {
+			result = isValidByte(value);
+		} else if (datatype == CoreDatatype.XSD_UNSIGNED_LONG) {
+			result = isValidUnsignedLong(value);
+		} else if (datatype == CoreDatatype.XSD_UNSIGNED_INT) {
+			result = isValidUnsignedInt(value);
+		} else if (datatype == CoreDatatype.XSD_UNSIGNED_SHORT) {
+			result = isValidUnsignedShort(value);
+		} else if (datatype == CoreDatatype.XSD_UNSIGNED_BYTE) {
+			result = isValidUnsignedByte(value);
+		} else if (datatype == CoreDatatype.XSD_FLOAT) {
+			result = isValidFloat(value);
+		} else if (datatype == CoreDatatype.XSD_DOUBLE) {
+			result = isValidDouble(value);
+		} else if (datatype == CoreDatatype.XSD_BOOLEAN) {
+			result = isValidBoolean(value);
+		} else if (datatype == CoreDatatype.XSD_DATETIME) {
+			result = isValidDateTime(value);
+		} else if (datatype == CoreDatatype.XSD_DATETIMESTAMP) {
+			result = isValidDateTimeStamp(value);
+		} else if (datatype == CoreDatatype.XSD_DATE) {
+			result = isValidDate(value);
+		} else if (datatype == CoreDatatype.XSD_TIME) {
+			result = isValidTime(value);
+		} else if (datatype == CoreDatatype.XSD_GDAY) {
+			result = isValidGDay(value);
+		} else if (datatype == CoreDatatype.XSD_GMONTH) {
+			result = isValidGMonth(value);
+		} else if (datatype == CoreDatatype.XSD_GMONTHDAY) {
+			result = isValidGMonthDay(value);
+		} else if (datatype == CoreDatatype.XSD_GYEAR) {
+			result = isValidGYear(value);
+		} else if (datatype == CoreDatatype.XSD_GYEARMONTH) {
+			result = isValidGYearMonth(value);
+		} else if (datatype == CoreDatatype.XSD_DURATION) {
+			result = isValidDuration(value);
+		} else if (datatype == CoreDatatype.XSD_DAYTIMEDURATION) {
+			result = isValidDayTimeDuration(value);
+		} else if (datatype == CoreDatatype.XSD_YEARMONTHDURATION) {
+			result = isValidYearMonthDuration(value);
+		} else if (datatype == CoreDatatype.XSD_QNAME) {
+			result = isValidQName(value);
+		} else if (datatype == CoreDatatype.XSD_ANYURI) {
+			result = isValidAnyURI(value);
+		} else if (datatype == CoreDatatype.XSD_LANGUAGE) {
+			result = Literals.isValidLanguageTag(value);
+		}
+
+		return result;
+	}
+
 	/**
 	 * Verifies if the supplied lexical value is a valid decimal or not.
 	 *
```

Comment:
Just wondering if a switch wouldn't be better/more readable.
I suppose it's just a matter of taste (assuming the JVM optimizes the code anyway)